app/javascript/src/components/Dashboard/Notes/NoteItemCard.jsx

```diff
@@ -0,0 +1,91 @@
+import React from "react";
+
+import { MenuVertical, Clock } from "@bigbinary/neeto-icons";
+import { Dropdown, Tag, Avatar, Tooltip } from "neetoui/v2";
+
+import { USR_IMG_TMP } from "../../Common/Sidebar/constants";
+
+//const { user } = useUserState();
+const getTimeSince = timestamp => {
+  var date = new Date(timestamp);
+  var seconds = Math.floor((new Date() - date) / 1000);
+  var interval = seconds / 31536000;
+
+  if (interval > 1) {
+    const floored = Math.floor(interval);
+    return +" year" + (floored > 1 ? "s" : "");
+  }
+  interval = seconds / 2592000;
+  if (interval > 1) {
+    const floored = Math.floor(interval);
+    return floored + " month" + (floored > 1 ? "s" : "");
+  }
+  interval = seconds / 86400;
+  if (interval > 1) {
+    const floored = Math.floor(interval);
+    return floored + " day" + (floored > 1 ? "s" : "");
+  }
+  interval = seconds / 3600;
+  if (interval > 1) {
+    const floored = Math.floor(interval);
+    return floored + " hour" + (floored > 1 ? "s" : "");
+  }
+  interval = seconds / 60;
+  if (interval > 1) {
+    const floored = Math.floor(interval);
+    return floored + " minute" + (floored > 1 ? "s" : "");
+  }
+  const floored = Math.floor(seconds);
+  return Math.floor(seconds) + " second" + (floored > 1 ? "s" : "");
+};
+
+const formatDate = timestamp => {
+  var options = { weekday: "long", hour: "2-digit", minute: "2-digit" };
+  return new Date(timestamp).toLocaleDateString("en-US", options);
+};
```

Comment:
1. Let's use `dayjs` here.
2. Suppose, we were going to use this, it's always better to extract such logic into a `helper.js` file.